sphinx_gallery/notebook.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+r"""
+============================
+Parser for Jupyter notebooks
+============================
+
+Class that holds the Ipython notebook information
+
+"""
+# Author: Óscar Nájera
+# License: 3-clause BSD
+
+from __future__ import division, absolute_import, print_function
+import json
+import os
+import re
+import sys
+
+def ipy_notebook_skeleton():
+    py_version = sys.version_info
+    notebook_skeleton = {
+        "cells": [],
+        "metadata": {
+            "kernelspec": {
+                "display_name": "Python " + str(py_version[0]),
+                "language": "python",
+                "name": "python" + str(py_version[0])
+            },
+            "language_info": {
+                "codemirror_mode": {
+                    "name": "ipython",
+                    "version": py_version[0]
+                },
+                "file_extension": ".py",
+                "mimetype": "text/x-python",
+                "name": "python",
+                "nbconvert_exporter": "python",
+                "pygments_lexer": "ipython" + str(py_version[0]),
+                "version": '{0}.{1}.{2}'.format(*sys.version_info[:3])
+            }
+        },
+        "nbformat": 4,
+        "nbformat_minor": 0
+    }
+    return notebook_skeleton
+
+
+def rst2md(text):
+    """Converts the RST text from the examples docstrigs and comments
+    into markdown text for the IPython notebooks"""
+
+    top_heading = re.compile(r'^=+$\s^([\w\s-]+)^=+$', flags=re.M)
+    text = re.sub(top_heading, r'# \1', text)
+
+    math_eq = re.compile(r'^\.\. math::((?:.+)?(?:\n+^  .+)*)', flags=re.M)
+    text = re.sub(math_eq,
+                  lambda match: r'$${0}$$'.format(match.group(1).strip()),
+                  text)
+    inline_math = re.compile(r':math:`(.+)`')
+    text = re.sub(inline_math, r'$\1$', text)
+
+    return text
+
```

Comment:
For PEP8 compatibility, you should have an extra empty line here (two empty lines between top-level blocks).